src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -55,7 +57,12 @@ createSQLiteStore dbFilename = do
 connectSQLiteStore :: MonadUnliftIO m => String -> m SQLiteStore
 connectSQLiteStore dbFilename = do
   dbConn <- liftIO $ DB.open dbFilename
-  return SQLiteStore {dbFilename, dbConn}
+  compileOptions <- liftIO (DB.query_ dbConn "pragma COMPILE_OPTIONS;" :: IO [[T.Text]])
+  let threadsafeOption = find (isPrefixOf "THREADSAFE=") (concat compileOptions)
+  case threadsafeOption of
+    Just "THREADSAFE=0" -> error "SQLite compiled with not threadsafe code" -- TODO warn instead
+    Just _ -> return SQLiteStore {dbFilename, dbConn} -- TODO info threadsafe
+    Nothing -> error "SQLite THREADSAFE compile option not found" -- should not happen
```

Comment:
Better put it into createSQLiteStore which is called once, connectSQLiteStore is called once per thread (twice for each TCP connection)

Reply:
If option not found - warn and continue
If option found and incorrect - we could exit or ask y/n to continue

Reply:
connectSQLiteStore +

I remember why I put it in createSQLiteStore - I was trying to be safe - but after giving it a second thought I see my reasoning was incorrect.

nvm this, I'll fix it